packages/elasticsearch-asset-apis/test/api-with-dataFrams-spec.ts

```diff
@@ -0,0 +1,484 @@
+            });
+
+            const results = await api.fetch() as DataFrame;
+
+            expect(results).toBeDefined();
+            expect(results.size).toEqual(1000);
+            expect(results.metadata.metrics).toMatchObject({
+                fetched: 1000,
+                total: 1000,
+                search_time: expect.any(Number)
+            });
+        });
+
+        it('can getWindowSize', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const size = await api.getWindowSize();
+
+            expect(size).toBeNumber();
+        });
+
+        it('can get api version', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const parsedNumber = toNumber(ELASTICSEARCH_VERSION.split('.')[0]);
+            expect(api.version).toEqual(parsedNumber);
+        });
+
+        // TODO this is badly named method, might need to change in the future
+        it('can verify index', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            let warnMessage: undefined|string;
+
+            const testLogger = {
+                info: () => {},
+                warn: (msg: string) => {
+                    warnMessage = msg;
+                    return null;
+                }
+            } as any;
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger: testLogger, emitter
+            });
+
+            await api.verifyIndex();
+
+            expect(warnMessage).toBeDefined();
+            expect(warnMessage).toBeString();
+        });
+
+        it('can make id slices', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const slicer = await api.makeIDSlicer({
+                slicerID: 0,
+                numOfSlicers: 1,
+                recoveryData: [],
+                keyType: IDType.base64url,
+                startingKeyDepth: 0,
+                idFieldName: 'uuid'
+            });
+
+            expect(slicer).toBeDefined();
+
+            const slice = await slicer() as SlicerDateResults;
+
+            expect(slice).toBeDefined();
+            expect(slice).toMatchObject({
+                wildcard: {
+                    field: 'uuid',
+                    value: 'a*'
+                },
+                count: 58
+            });
+        });
+    });
+
+    describe('returning data entities', () => {
```

Comment:
Why does this file include tests for data frame results, shouldn't it just be for data frames?